### ISSUE TITLE
Empty "accessibility caption" in data collection from a profile

### ISSUE THREAD

**Issue by pvita** [instaloader/instaloader#1299]
I'm collecting metadata Jsons from a profile and the "accessibilty_caption" values are 99% null, also if these data are clearly visible in the source pages from the website.

**To Reproduce**
instaloader profile --metadata-json 

**Instaloader version**
4.8.1

**Comment by pvita** [comment]
It seems that Instaloader collect properly only the last days metadata. Probably the ones from the posts present on the first loading of the profile.

**Comment by pvita** [comment]
Metadata are properly collected when I make a query to a single shortcode

**Comment by aandergr** [comment]
The JSON files Instaloader saves along with each Post contain all the metadata that has been retrieved from Instagram while downloading the picture and associated required information. However, Instaloader tries to do as few metadata requests as possible, so, depending on how Instaloader has been invoked, it may occur that these files do not contain the complete available metadata structure.

**Comment by pvita** [comment]
Thanks for your answer. But the problem is that "accessibility_caption" are (after the first 10 more or less) empty if you're collecting data from a profile.
So I had to set a list of shortcodes as target. 

For instance:
Command: instaloader joebiden --meta-data-json
If your target is "joebiden" this shortcode **CTz-IAeJMFC** has an empty accessibility caption

Command: instaloader -- -CTz-IAeJMFC --meta-data-json
If your target is CTz-IAeJMFC this is the accessibility caption "Photo shared by Joe Biden on September 14, 2021 tagging @gavinnewsom. May be an image...."

I don't know why, maybe it is a limitation on the politicians' profiles.
I didn't try with other categories.

**Comment by aandergr** [comment]
Yes, the JSON files saved by Instaloader may not contain all information. This behavior is intentional, as it saves requests.

However, the script snippet that I posted in my previous comment allows you to obtain the accessibility_caption of all posts where it is available.

**Comment by pvita** [comment]
Hi @aandergr is there any chance to run this cmd command 
"instaloader --comments"  with a folder as target, in order to download all the comments from all the posts?

And the same for "--metadata-json" if it is possibile...
